scripts/templates/valddi.cpp.mako

```diff
@@ -114,15 +127,10 @@ namespace ur_validation_layer
             }
         }
         %elif func_name in tp_handle_funcs['retain']:
-        if( getContext()->enableLeakChecking && result == UR_RESULT_SUCCESS )
```

Comment:
Instead of counting the number of successful retains/frees, we count the number of retain/frees in total, regardless of their success. I'm not sure there's a clear way to update the reference count after actually freeing the handle. And I'm also not sure reference count checking would be meaningful in an environment where retain/free are failing.